app/feedme/versioning_service.py

```
@@ -0,0 +1,427 @@
+                return ConversationVersion(**dict(new_version_row))
+                
+        except Exception as e:
+            conn.rollback()
+            logger.error(f"Failed to create new version for conversation {conversation_id}: {e}")
+            raise
+    
+    def get_conversation_versions(self, conversation_id: int) -> VersionListResponse:
+        """Get all versions of a conversation"""
+        try:
+            with self.connection_manager.get_connection() as conn:
+                with conn.cursor(cursor_factory=psycopg2_extras.RealDictCursor) as cur:
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)